src/plugins/credential_management/public/components/create_credential_wizard/create_credential_wizard.tsx

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import {
+  EuiHorizontalRule,
+  EuiGlobalToastList,
+  EuiGlobalToastListToast,
+  EuiForm,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSelect,
+  EuiLink,
+  EuiButton,
+  EuiPageContent,
+  EuiFieldPassword,
+} from '@elastic/eui';
+import { DocLinksStart } from 'src/core/public';
+import { getCreateBreadcrumbs } from '../breadcrumbs';
+import { CredentialManagmentContextValue } from '../../types';
+import { Header } from './components/header';
+import { context as contextType } from '../../../../opensearch_dashboards_react/public';
+
+interface CreateCredentialWizardState {
+  credentialName: string;
+  credentialType: string;
+  userName: string;
+  password: string;
+  dual: boolean;
+  toasts: EuiGlobalToastListToast[];
+  docLinks: DocLinksStart;
+}
+
+export class CreateCredentialWizard extends React.Component<
+  RouteComponentProps,
+  CreateCredentialWizardState
+> {
+  static contextType = contextType;
+  public readonly context!: CredentialManagmentContextValue;
+  constructor(props: RouteComponentProps, context: CredentialManagmentContextValue) {
+    super(props, context);
+
+    context.services.setBreadcrumbs(getCreateBreadcrumbs());
+
+    this.state = {
+      credentialName: '',
+      credentialType: 'username_password_credential',
+      userName: '',
+      password: '',
+      dual: true,
+      toasts: [],
+      docLinks: context.services.docLinks,
+    };
+  }
+
+  // TODO: Fix header component error https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2048
+  renderHeader() {
+    const { docLinks } = this.state;
+
+    return <Header docLinks={docLinks} />;
+  }
+
+  renderContent() {
+    const header = this.renderHeader();
+
+    const options = [
+      { value: 'username_password_credential', text: 'Username and Password Credential' },
+      { value: 'no_auth', text: 'No Auth' },
```

Comment:
auth types will be used on both browser and server side, we may want to define supported auth types in `common` dir and try reuse whatever possible

Reply:
Yes, since this PR only focused on browser side changes, plus the refactor suggestion https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2040#discussion_r936123647. Maybe we can add the common model in the new plugin.